### ISSUE TITLE
Fix leaky test

### ISSUE THREAD

**Issue by peterschutt** [starlite-api/starlite#502]
Seems like we have a test that is leaking out logging config changes to other tests.

Run `$ poetry run pytest tests examples` which is the command we run in the 3.10 CI tests and we get warnings that originate out of `picologging/handlers.py`:

```python-traceback
============================================================================================== warnings summary ==============================================================================================
examples/tests/test_using_application_state.py::test_using_application_state
  /home/peter/.cache/pypoetry/virtualenvs/starlite-CEGnKvDD-py3.10/lib/python3.10/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-8 (_monitor)
  
  Traceback (most recent call last):
    File "/home/peter/.pyenv/versions/3.10.6/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
      self.run()
    File "/home/peter/.pyenv/versions/3.10.6/lib/python3.10/threading.py", line 953, in run
      self._target(*self._args, **self._kwargs)
    File "/home/peter/.cache/pypoetry/virtualenvs/starlite-CEGnKvDD-py3.10/lib/python3.10/site-packages/picologging/handlers.py", line 621, in _monitor
      self.handle(record)
    File "/home/peter/.cache/pypoetry/virtualenvs/starlite-CEGnKvDD-py3.10/lib/python3.10/site-packages/picologging/handlers.py", line 602, in handle
      handler.handle(record)
  TypeError: Argument must be a LogRecord
  
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

examples/tests/application_hooks/test_application_after_exception_hook.py::test_application_shutdown_hooks
  /home/peter/.cache/pypoetry/virtualenvs/starlite-CEGnKvDD-py3.10/lib/python3.10/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-11 (_monitor)
  
  Traceback (most recent call last):
    File "/home/peter/.pyenv/versions/3.10.6/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
      self.run()
    File "/home/peter/.pyenv/versions/3.10.6/lib/python3.10/threading.py", line 953, in run
      self._target(*self._args, **self._kwargs)
    File "/home/peter/.cache/pypoetry/virtualenvs/starlite-CEGnKvDD-py3.10/lib/python3.10/site-packages/picologging/handlers.py", line 621, in _monitor
      self.handle(record)
    File "/home/peter/.cache/pypoetry/virtualenvs/starlite-CEGnKvDD-py3.10/lib/python3.10/site-packages/picologging/handlers.py", line 602, in handle
      handler.handle(record)
  TypeError: Argument must be a LogRecord
  
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Then run the same tests but swap the order between `examples` and `tests`, so: `$ poetry run pytest examples tests` and we get no warnings. Same if you run the example tests alone, `$ poetry run pytest examples` - no warnings.

**Comment by peterschutt** [comment]
It's this test in `test_picologging.py`:

```python
def test_logger_startup(caplog: "LogCaptureFixture") -> None:
    with TestClient(
        app=Starlite(
            route_handlers=[],
            on_startup=[
                LoggingConfig(
                    handlers={
                        "console": {
                            "class": "picologging.StreamHandler",
                            "level": "DEBUG",
                            "formatter": "standard",
                        },
                        "queue_listener": {
                            "class": "starlite.logging.picologging.QueueListenerHandler",
                            "handlers": ["cfg://handlers.console"],
                        },
                    }
                ).configure
            ],
        )
    ) as client, caplog.at_level(logging.INFO):
        client.options("/")
        test_logger = logging.getLogger()
        handlers = test_logger.handlers

        test_picologging_logger = picologging.getLogger()
        test_picologging_handlers = test_picologging_logger.handlers
        assert isinstance(handlers[0].handlers[0], picologging.StreamHandler)  # type: ignore
        assert isinstance(test_picologging_handlers[0].handlers[0], picologging.StreamHandler)
```
@cofin ping

**Comment by peterschutt** [comment]
Also @cofin, there is this stuff defined in module scope in `test_picologging.py`:

https://github.com/starlite-api/starlite/blob/c9387f8c24b0e66fadbceecf05f3611b76f6b7bf/tests/logging_config/test_picologging.py#L106-L120

If I delete it, all the tests still pass, does it need to be there? If so, maybe some comments would be useful.

**Comment by peterschutt** [comment]
Closed in 55f590703823acfaed252d3b37610e690423b57d
